### ISSUE TITLE
field-bitmap plugin broken by Blockly v9 due to `TypeError` in `FieldBitmap` constructor

### ISSUE THREAD

**Issue by cpcallen** [rachel-fenichel/blockly-samples#1268]
**Category**

 - Plugins

**Component**

field-bitmap

**Describe the bug**

`TypeError: Cannot read properties of undefined (reading 'length')` in `FieldBitmap` constructor 

**To Reproduce**

1. Go to `plugins/field-bitmap`
2. Run `npm install blockly@9.0.0-beta.1`
3. Run `npm start`
   * Or skip steps 1–3 and go to [the demo on GitHub Pages](https://google.github.io/blockly-samples/plugins/field-bitmap/test/)).
5. Open browser console.
6. Observe demo seems not to have loaded correctly; see `TypeError` in console (see stack trace below).

**Expected behaviour**

No error; demo loads properly.

**Screenshots**

This doesn't look correct to me: where is the toolbox / grid / trashcan?:

<img width="823" alt="Screenshot 2022-09-10 at 20 27 30" src="https://user-images.githubusercontent.com/5603958/189498918-3080e1f1-f4bd-4513-ac0c-52268e1cdfd4.png">

**Stack traces**

```
TypeError: Cannot read properties of undefined (reading 'length')
    at new FieldBitmap (field-bitmap.js?3b22:39:25)
    at Function.fromJson (field-bitmap.js?3b22:78:12)
    at Object.fromJsonInternal$$module$build$src$core$field_registry [as fromJsonInternal] (field_registry.ts?42f1:82:22)
    at fromJson$$module$build$src$core$field_registry (field_registry.ts?42f1:61:20)
    at BlockSvg$$module$build$src$core$block_svg.fieldFromJson_ (block.ts?6432:1813:33)
    at BlockSvg$$module$build$src$core$block_svg.interpolate_ (block.ts?6432:1713:28)
    at BlockSvg$$module$build$src$core$block_svg.jsonInit (block.ts?6432:1554:12)
    at BlockSvg$$module$build$src$core$block_svg.eval [as init] (common.ts?52f0:237:10)
    at BlockSvg$$module$build$src$core$block_svg.doInit_ (block.ts?6432:287:14)
    at new BlockSvg$$module$build$src$core$block_svg (block_svg.ts?8904:208:10)
```

**Comment by rachel-fenichel** [comment]
The proximate cause is that field's value is `undefined` after the super constructor if no value was passed in. There's code to check for `null`, but not `undefined`, so it tries to index into the value and crashes.

The other issue is that it's not using the `DEFAULT_VALUE` from the base field class. If it were, value would be `null` instead of `undefined`. I think that's the piece that changed since the last release.

`field.ts` sets the default value like this:
```
  /** The default value for this field. */
  protected DEFAULT_VALUE: any = null;
```

and uses it like this:
```
    /**
     * A generic value possessed by the field.
     * Should generally be non-null, only null when the field is created.
     */
    this.value_ = (new.target).prototype.DEFAULT_VALUE;
```

**Comment by rachel-fenichel** [comment]
After discussion with Beka and Christopher:
- The default value should be set as `this.value_ = 'DEFAULT_VALUE' in (new.target).prototype ? (new.target).prototype.DEFAULT_VALUE : null;` so that it's `null` if it was not set on the prototype object.
- Beka will make that change in core
- Rachel will merge a few open PRs into the v9 branch to avoid rebases, then publish a beta from core and update the v9 branch to use the new beta
- Rachel will continue to test and fix plugins

**Comment by rachel-fenichel** [comment]
Fixed by google/blockly/#6453
